host/crypto/openssl/rsa.c

```diff
@@ -14,6 +14,60 @@
 #include "key.h"
 #include "rsa.h"
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+/* Needed for compatibility with ssl1.1 */
+
+static void RSA_get0_key(
+    const RSA* r,
+    const BIGNUM** n,
+    const BIGNUM** e,
+    const BIGNUM** d)
+{
+    if (n != NULL)
+        *n = r->n;
+    if (e != NULL)
+        *e = r->e;
+    if (d != NULL)
+        *d = r->d;
+}
+
+static int RSA_set0_key(RSA* r, BIGNUM* n, BIGNUM* e, BIGNUM* d)
+{
+    /* If the fields n and e in r are NULL, the corresponding input
+     * parameters MUST be non-NULL for n and e.  d may be
+     * left NULL (in case only the public key is used).
+     */
+    if ((r->n == NULL && n == NULL) || (r->e == NULL && e == NULL))
+        return 0;
+
+    if (n != NULL)
+    {
+        BN_free(r->n);
+        r->n = n;
+    }
+    if (e != NULL)
+    {
+        BN_free(r->e);
+        r->e = e;
+    }
+    if (d != NULL)
+    {
+        BN_free(r->d);
+        r->d = d;
+    }
+
+    return 1;
+}
+
+#else
```

Comment:
Do we need these forward definitions? I expect that these are part of the 1.1 header definitions already (i.e. we don't have to do this in the other headers, why here?)

Reply:
They are part of the 1.1 but not 1.0 header. We need them for 16.04

Reply:
Can we add a comment to that effect?

Reply:
Can indeed.

Reply:
I removed the forward decls. It still compiles